app/geo/gui/src/ts/twxplore/gui/geo/attributeStrategies/classes/NumericFeatureAttribute.ts

```diff
@@ -0,0 +1,52 @@
+import {FeatureAttribute} from "./FeatureAttribute";
+import {MapFeatureAttributeState} from "../../states/map/MapFeatureAttributeState/MapFeatureAttributeState";
+import {setInitialFiltersNumeric} from "../functions/setInitialFilters/setInitialFiltersNumeric";
+import {FeatureAttributeName} from "../../states/map/FeatureAttributeName";
+import {TypeOfFeatureAttribute} from "../../states/map/TypeOfFeatureAttribute";
+import {KeplerFilterType} from "../../states/map/KeplerFilterType";
+import {KeplerFieldType} from "../../states/map/KeplerFieldType";
+import {Dispatch} from "redux";
+import {buildInitialFeatureAttributeStateNumeric} from "../functions/buildInitialFeatureAtttributeState/buildInitialFeatureAttributeStateNumeric";
+import {MapFeature} from "../../states/map/MapFeature";
+import {updateNumericAttributeStateOfFeatureType} from "../functions/updateAttributeStateOfFeatureType/updateNumericAttributeStateOfFeatureType";
+
+export abstract class NumericFeatureAttribute implements FeatureAttribute {
+  updateAttributeStatesOfFeatureType(
+    attributeStatesOfFeatureType: {
+      [featureAttributeName: string]: MapFeatureAttributeState;
+    },
+    addedFeature: MapFeature
+  ): void {
+    updateNumericAttributeStateOfFeatureType(
+      attributeStatesOfFeatureType,
+      addedFeature,
+      this.name
+    );
+  }
+  buildInitialFeatureAttributeState(attributeStatesOfFeatureType: {
+    [featureAttributeName: string]: MapFeatureAttributeState;
+  }): void {
+    buildInitialFeatureAttributeStateNumeric(
+      attributeStatesOfFeatureType,
+      this.name
+    );
+  }
+  setInitialFilters(
+    filterIndexOfAttribute: number,
+    stateOfAttribute: MapFeatureAttributeState,
+    dispatch: Dispatch<any>
+  ): void {
+    setInitialFiltersNumeric(
+      filterIndexOfAttribute,
+      this.name,
+      stateOfAttribute,
+      this.KeplerFilterType,
+      dispatch
+    );
+  }
+  abstract readonly name: FeatureAttributeName;
```

Comment:
I don't think you need to redeclare these if they're in FeatureAttribute?

Reply:
It seems so. Else the class won't implement FeatureAttribute as declared.